### PR TITLE
feat: add recycle bin endpoints

### LINKED ISSUE #66: Add recycle bin endpoints

### PR DESCRIPTION
**Pull request by pyphilia**
close #66

depends on https://github.com/graasp/graasp/pull/79

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export type Member = {
   id: UUID;
   name: string;
   email: string;
-  extra: {};
+  extra: any;
```
Comment (Julien-Torrent) [inline src/types.ts:37]: I think it is better to use `unknown`

File: test/utils.tsx [hunk 2]
```diff
@@ -14,6 +14,34 @@ import configureQueryClient from '../src/queryClient';
...
+//   isPlaceholderData: false,
+//   isPreviousData: false,
+//   isStale: true,
+//   refetch: () =>
+//     new Promise<QueryObserverResult<Map<unknown, unknown>, unknown>>(
+//       () => true,
+//     ),
+//   remove: () => {},
+// };
+
```
Comment (Julien-Torrent) [inline test/utils.tsx:44]: If this is unused, we should delete it.

**Review by Julien-Torrent — CHANGES_REQUESTED** [review]

File: test/utils.tsx (hunk 2 above)
Comment (juancarlosfarah) [reply test/utils.tsx:44]: Agree.

File: src/mutations/item.ts [hunk 3]
```diff
@@ -180,20 +182,121 @@ export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
...
+      const previousItems = {
+        ...(Boolean(itemPath) && {
+          parent: await mutateParentChildren({
+            childPath: itemPath,
+            value: (old: List<Item>) =>
+              old.filter(({ id }) => !itemIds.includes(id)),
+          }),
+        }),
+      };
+      // items themselves still exists but the path is different
```
Comment (juancarlosfarah) [inline src/mutations/item.ts:253]: exist*

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Looks good! Some minor comments. Agree with @Julien-Torrent. 👍

**Pull request merged**